src/spi.rs

```diff
@@ -676,6 +689,42 @@ impl<SPI: Instance, PINS, const BIDI: bool, W: FrameSize, OPERATION>
             nb::Error::WouldBlock
         })
     }
+
+    // Implement write as per the "Transmit only procedure"
+    // RM SPI::3.5. This is more than twice as fast as the
+    // default Write<> implementation (which reads and drops each
+    // received value)
+    fn spi_write<WI>(&mut self, words: WI) -> Result<(), Error>
+    where
+        WI: IntoIterator<Item = W>,
+    {
+        if BIDI {
+            self.spi.cr1.modify(|_, w| w.bidioe().set_bit());
+        }
+        // Write each word when the tx buffer is empty
+        for word in words {
+            loop {
+                let sr = self.spi.sr.read();
+                if sr.txe().bit_is_set() {
+                    self.write_data_reg(word);
+                    if sr.modf().bit_is_set() {
+                        return Err(Error::ModeFault);
+                    }
+                    break;
+                }
+            }
+        }
+        // Wait for final TXE
+        while !self.is_tx_empty() {}
+        // Wait for final !BSY
+        while self.is_busy() {}
```

Comment:
This is 1 moment I'm not sure. Do we need to check BSY here?
It takes 3-4 SPI clocks.

Reply:
I moved `busy` check to `Spi::disable`